src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -8,6 +8,11 @@ internal static class ServiceFeatureOnOffOptions
     {
         public const string OptionName = "ServiceFeaturesOnOff";
 
+        /// <summary>
+        /// this option is solely for performance. don't confused by option name. 
+        /// this option doesn't mean we will show all diagnostics that belong to opened files when turned off,
+        /// rather it means we will only show diagnostics that are cheap to calculate for small scope such as opened files.
+        /// </summary>
         public static readonly PerLanguageOption<bool> ClosedFileDiagnostic = new PerLanguageOption<bool>(OptionName, "Closed File Diagnostic", defaultValue: true);
```

Comment:
I meant like this
http://index/#Microsoft.VisualStudio.LanguageServices.CSharp/Options/AutomationObject.cs,39

or when it persisted, it will use the "Closed File Diagnostic". if we change it, previously saved option will be lost.